external-crates/move/crates/move-compiler/src/hlir/translate.rs

```diff
@@ -373,6 +244,44 @@ impl<'env> Context<'env> {
     }
 }
 
+impl MatchContext<true> for Context<'_> {
+    fn env(&mut self) -> &mut CompilationEnv {
+        self.env
+    }
+
+    fn env_ref(&self) -> &CompilationEnv {
+        self.env
+    }
+
+    /// Makes a new `naming/ast.rs` variable. Does _not_ record it as a function local, since this
+    /// should only be called in match compilation, which will have its body processed in HLIR
+    /// translation after expansion.
+    fn new_match_var(&mut self, name: String, loc: Loc) -> N::Var {
+        let id = self.counter_next();
+        let name = format!(
+            "{}{NEW_NAME_DELIM}{name}{NEW_NAME_DELIM}{id}",
```

Comment:
Curious if you want to share the definition of this (modulo the counter) in `shared` or not. Otherwise I worry the formatting of match temps could diverge from each other possibly?

Reply:
Good point! I moved name generation to `shared/matching.rs`.